### PR TITLE
Update JSONRPC-trace-module.md

### PR DESCRIPTION
**Pull request by liamaharon**
Resolves https://github.com/paritytech/parity-ethereum/issues/11365

### CODE REVIEW

**Comment by liamaharon** [comment]
[clabot:check]

**Comment by joshua-mir** [comment]
@liamaharon thanks for the PR! I'm also fairly sure that integers used to work at some point (as per your discussion in the linked issue) but all indications in the codebase indicate there haven't been any changes to this logic for years 😅

**Comment by niklasad1** [comment]
@joshua-mir are you implying that we should fix the code instead?

There are many more places where the documentation needs to be fixed i.e, each RPC-type that has at least one `U256` field

**Comment by joshua-mir** [comment]
it would be nice to know if more people have run into this change, or when exactly it happened. fwiw nobody has been complaining about it, other than general requests for knowing what types our rpc apis are meant to use, and these issues are the first time I'm hearing of a difference in behavior from what people expect.

File: JSONRPC-trace-module.md [hunk 1]
```diff
@@ -74,9 +74,9 @@ Executes the given call and returns a number of possible traces for it.
 0. `Object` - Call options, same as `eth_call`.
     - `from`:   `Address` - (optional) 20 Bytes - The address the transaction is send from.
     - `to`:   `Address` - (optional when creating new contract) 20 Bytes - The address the transaction is directed to.
-    - `gas`:   `Quantity` - (optional) Integer of the gas provided for the transaction execution. eth_call consumes zero gas, but this parameter may be needed by some executions.
-    - `gasPrice`:   `Quantity` - (optional) Integer of the gas price used for each paid gas.
-    - `value`:   `Quantity` - (optional) Integer of the value sent with this transaction.
+    - `gas`:   `Quantity` - (optional) Hex string of the gas provided for the transaction execution. eth_call consumes zero gas, but this parameter may be needed by some executions.
```
Comment (dvdplm) [inline JSONRPC-trace-module.md]: I guess it's still an integer, but expressed in hex format, which for the JSON case means a hex string?

```suggestion
    - `gas`:   `Quantity` - (optional) Integer formatted as a hex string of the gas provided for the transaction execution. eth_call consumes zero gas, but this parameter may be needed by some executions.
```

Maybe?

Comment (liamaharon) [reply JSONRPC-trace-module.md]: Good suggestion! I've updated the branch

**Comment by niklasad1** [comment]
> it would be nice to know if more people have run into this change, or when exactly it happened. fwiw nobody has been complaining about it, other than general requests for knowing what types our rpc apis are meant to use, and these issues are the first time I'm hearing of a difference in behavior from what people expect.

This behavior dates back to at least ~2016, see https://github.com/paritytech/parity-ethereum/pull/2247/files. I think it actually been there for so long so everyone got used to it but I don't know to be honest.

**Pull request merged**